tests/resources/test_permissions.py

```diff
@@ -0,0 +1,185 @@
+from unittest.mock import MagicMock
+from datetime import datetime
+from typing import Tuple
+
+from cidc_api.models import (
+    Users,
+    Permissions,
+    PermissionSchema,
+    TrialMetadata,
+    CIDCRole,
+)
+
+from ..utils import mock_current_user, make_admin
+
+TRIAL_ID = "foo"
+
+
+def setup_permissions(cidc_api, monkeypatch) -> Tuple[int, int]:
+    """
+    Create two users, one trial, and three permissions in `db`.
+    Two permissions will belong to the first user, and the third will
+    belong to the second one. Returns the first and second user ids 
+    as a tuple.
+    """
+    current_user = Users(
+        id=1,
+        email="test@email.com",
+        role=CIDCRole.CIMAC_USER.value,
+        approval_date=datetime.now(),
+    )
+    other_user = Users(id=2, email="other@email.org")
+
+    mock_current_user(current_user, monkeypatch)
+
+    with cidc_api.app_context():
+        # Create users
+        current_user.insert()
+        other_user.insert()
+
+        # Create trial
+        TrialMetadata.create(TRIAL_ID, {})
+
+        # Create permissions
+        def create_permission(uid, assay):
+            Permissions(
+                granted_by_user=uid,
+                granted_to_user=uid,
```

Comment:
nit: not with regard to this code, but shouldn't granting perms to a user by themselves be, like, an error?

Reply:
Hmm, maybe - kind of a RBAC question, though, and right now that's enforced by the `/permissions` endpoint, which handles that issue by only allowing admins to grant permissions, rather than the `Permissions` model.